pkg/kubectl/resource/builder.go

```diff
@@ -290,24 +291,40 @@ func (b *Builder) ResourceNames(resource string, names ...string) *Builder {
 	return b
 }
 
-// SelectorParam defines a selector that should be applied to the object types to load.
+// LabelSelectorParam defines a selector that should be applied to the object types to load.
 // This will not affect files loaded from disk or URL. If the parameter is empty it is
 // a no-op - to select all resources invoke `b.Selector(labels.Everything.String)`.
-func (b *Builder) SelectorParam(s string) *Builder {
+func (b *Builder) LabelSelectorParam(s string) *Builder {
 	selector := strings.TrimSpace(s)
 	if len(selector) == 0 {
 		return b
 	}
 	if b.selectAll {
-		b.errs = append(b.errs, fmt.Errorf("found non empty selector %q with previously set 'all' parameter. ", s))
+		b.errs = append(b.errs, fmt.Errorf("found non-empty label selector %q with previously set 'all' parameter. ", s))
 		return b
 	}
-	return b.Selector(selector)
+	return b.LabelSelector(selector)
 }
 
-// Selector accepts a selector directly, and if non nil will trigger a list action.
-func (b *Builder) Selector(selector string) *Builder {
-	b.selector = &selector
+// LabelSelector accepts a labelSelector directly, and if non nil will trigger a list action.
```

Comment:
Godoc is wrong now, selector can't be nil.  `LabelSelector accepts a selector directly and will filter the resulting list by that object.  Use LabelSelectorParam instead for user input.`